query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -430,6 +449,16 @@ public Observable<Row> handleRequest(
     }
   }
 
+  private void measureRequestTimeRangeDuration(Duration timeRangeDuration) {
+    try {
+      long minutes = convertSecondsToMinutes(timeRangeDuration.getSeconds());
+      LOG.info("Pinot {} : timeRangeDuration of request in minutes: {}.", name, minutes);
```

Comment:
TLDR: Make this log structured.

As a general rule, let us not use placeholders for logging. There are two downsides to this:
1. Logs would come as "Pinot _name_  : timeRangeDuration of request in minutes: 15". Now we'll have to extract _15_ from this log.
2. Sumo will create a full-text index for this entire log, which increases load on it.

If we go with structured logging, we'll have something like:

log: `{LEVEL: INFO, bookmark: "QUERY_TIME_SPAN", "duration": 15}`

Because it's JSON formatted, we won't have to extract anything. Search for this log can be made using the bookmark.

[Ref](https://medium.com/@zachcorbettmcelrath/structured-logging-in-spring-boot-with-log4j2-part-2-logging-objects-7fa77f8a8c8c#:~:text=Yep!%20We%20can%20do%20that%20instead%3A)

Reply:
Done